workflow/controller/operator.go

```diff
@@ -307,11 +307,17 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 		woc.wf.Status.EstimatedDuration = woc.estimateWorkflowDuration()
 	} else {
 		woc.workflowDeadline = woc.getWorkflowDeadline()
-		err = woc.podReconciliation(ctx)
+		err, podReconciliationCompleted := woc.podReconciliation(ctx)
 		if err == nil {
 			woc.failSuspendedAndPendingNodesAfterDeadlineOrShutdown()
 		}
 
+		if !podReconciliationCompleted {
+			woc.log.WithField("workflow", woc.wf.ObjectMeta.Name).Info("pod reconciliation didn't complete, will retry")
+			woc.requeue()
+			return
+		}
```

Comment:
Sorry, I just realized that we probably need to move the `if err != nil` clause above the `if !podReconciliationCompleted {`, since we can return `err, false`

Reply:
hopefully after that we should be good, thank you for the iterations!

Reply:
Yes, I changed it.